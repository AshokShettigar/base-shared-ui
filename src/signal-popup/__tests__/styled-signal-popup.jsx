import { shallow } from 'enzyme';
import StyledSignalPopup from 'platform-common-ui/signal-popup/styled-signal-popup';

describe('platform-common-ui/signal-popup/styled-signal-popup', function() {
    const props = {
        data: {
            ticketsCnt: 10
        },
        popupConfig: {
            Component: ''
        },
        buttonConfig: {
            buttonLabelField: 'ticketsCnt',
            className: 'wrapper',
            iconName: 'critical-alert'
        },
        colDef: {
            cellRendererParams: {
                LayoutComponent: <div />
            }
        },
        position: 'top-end'
    };

    it('should render the StyledSignalPopup', function() {
        const wrapper = shallow(
            <StyledSignalPopup
                {...props}
                onPositionChanged={jest.fn()}
            />
            );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render blank div if Component is not passed', function() {
        const newProps = {
            ...props,
            popupConfig: {}
        };

        const wrapper = shallow(
            <StyledSignalPopup
                {...newProps}
                onPositionChanged={jest.fn()}
            />
            );
        expect(wrapper).toMatchSnapshot();
    });

    it('should call updatePositionChanged when new position is passed', function() {
        const mockUpdatePositionChanged = jest.fn().mockImplementationOnce(calledWith => {
            expect(calledWith).toEqual('bottom-end');
        });
        const newProps = {
            ...props,
            popupConfig: {}
        };

        const wrapper = shallow(
            <StyledSignalPopup
                {...newProps}
                onPositionChanged={mockUpdatePositionChanged}
            />
            );
        wrapper.instance().componentWillReceiveProps({ position: 'bottom-end' });
    });

    it('should call updatePosition when component updates', function() {
        const mockUpdatePosition = jest.fn();
        const newProps = {
            ...props,
            popupConfig: {}
        };

        const wrapper = shallow(
            <StyledSignalPopup
                {...newProps}
                onPositionChanged={jest.fn()}
                updatePosition={mockUpdatePosition}
            />
            );
        wrapper.instance().componentDidUpdate();
        expect(mockUpdatePosition).toHaveBeenCalled();
    });
});
