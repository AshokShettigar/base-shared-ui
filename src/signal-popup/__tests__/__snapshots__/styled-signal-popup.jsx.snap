// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`platform-common-ui/signal-popup/styled-signal-popup should render blank div if Component is not passed 1`] = `
<div
  data-component="StyledSignalPopup"
/>
`;

exports[`platform-common-ui/signal-popup/styled-signal-popup should render the StyledSignalPopup 1`] = `
<div
  data-component="StyledSignalPopup"
/>
`;
